psi4/driver/qcdb/interface_gcp.py

```diff
@@ -0,0 +1,307 @@
+    except OSError as e:
+        raise ValidationError('Program gcp not found in path. %s' % e)
+    out, err = dashout.communicate()
+
+    # Parse output
+    success = False
+    for line in out.splitlines():
+        line = line.decode('utf-8')
+        if re.match('  Egcp:', line):
+            sline = line.split()
+            dashd = float(sline[1])
+        if re.match('     normal termination of gCP', line):
+            success = True
+
+    if not success:
+        os.chdir(current_directory)
+        raise Dftd3Error("""Unsuccessful gCP run.""")
+
+    # Parse grad output
+    if dertype != 0:
+        derivfile = './gcp_gradient'
+        dfile = open(derivfile, 'r')
+        dashdderiv = []
+        for line in geom.splitlines():
+            lline = line.split()
+            if len(lline) != 4:
+                continue
+            if lline[0] == 'Gh':
+                dashdderiv.append([0.0, 0.0, 0.0])
+            else:
+                dashdderiv.append([float(x.replace('D', 'E')) for x in dfile.readline().split()])
+        dfile.close()
+
+        if len(dashdderiv) != self.natom():
+            raise ValidationError('Program gcp gradient file has %d atoms- %d expected.' % \
+                (len(dashdderiv), self.natom()))
+
+    # Prepare results for Psi4
+    if isP4regime and dertype != 0:
+        core.set_variable('GCP CORRECTION ENERGY', dashd)
+        psi_dashdderiv = core.Matrix(self.natom(), 3)
+        psi_dashdderiv.set(dashdderiv)
+
+    # Print program output to file if verbose
+    if not verbose and isP4regime:
+        verbose = True if core.get_option('SCF', 'PRINT') >= 3 else False
+    if verbose:
+
+        text = '\n  ==> GCP Output <==\n'
+        text += out.decode('utf-8')
+        if dertype != 0:
+            with open(derivfile, 'r') as handle:
+                text += handle.read().replace('D', 'E')
+            text += '\n'
+        if isP4regime:
+            core.print_out(text)
+        else:
+            print(text)
+
+#    # Clean up files and remove scratch directory
+#    os.unlink(paramfile1)
+#    os.unlink(paramfile2)
+#    os.unlink(geomfile)
+#    if dertype != 0:
+#        os.unlink(derivfile)
+#    if defmoved is True:
+#        os.rename(defaultfile + '_hide', defaultfile)
+
+    os.chdir('..')
+#    try:
+#        shutil.rmtree(dftd3_tmpdir)
+#    except OSError as e:
+#        ValidationError('Unable to remove dftd3 temporary directory %s' % e)
+    os.chdir(current_directory)
+
+    # return -D & d(-D)/dx
+    if dertype == -1:
+        return dashd, dashdderiv
+    elif dertype == 0:
+        return dashd
+    elif dertype == 1:
+        return psi_dashdderiv
+
+try:
+    # Attach method to libmints psi4.Molecule class
+    core.Molecule.run_gcp = run_gcp
+except (NameError, AttributeError):
+    # But don't worry if that doesn't work b/c
+    #   it'll get attached to qcdb.Molecule class
+    pass
```

Comment:
What I mean is specifically in the future a superfunctional is really a tuple of (Superfunctional, EmpericalDispersion). The EmpericalDispersion object can be duck typed so that it doesnt really matter what it it is along as you follow `compute_energy`, `compute_gradient`, etc. However, adding GCP as an option to EmpericalDispersion would be a favorable route as it already contains psi4.core.Dispersion and dftd3. Its just a convenient place to put more atom pairwise corrections without having to add them anywhere else.